15314f1ddb882700dc9fabc5ca961943/update/sys_script_include_b630a09ddb9020d01206fd641d9619df.xml

```diff
@@ -718,20 +718,355 @@ IllumioUtils.prototype = {
+	 * @param {String} pceConfig object of configuration
+	 * @returns {dict} Dict containing the status, response and response status code
+	 */
+	getLabelDimensions: function(pceConfig) {
+		try {
+			var returnDict = {
+				"status": false,
+				"response": "",
+				"statusCode": -1
+			};
+
+			if (!(pceConfig.mid_server.status == MID_SERVER_UP_STATUS && pceConfig.mid_server.validated == 'true')) {
+				gs.error("The configured MID Server with configuration record " + pceConfig.getUniqueValue() + " is down.");
+				return returnDict;
+			}
+			
+			var stringParams = {
+				"url": pceConfig.pce_url + "",
+				"orgID": (pceConfig.organization_id || "1") + ""
+			};
+			var pce_authorization = gs.base64Encode(pceConfig.username.getDecryptedValue() + ":" + pceConfig.secret_key.getDecryptedValue());
+			var headers = {
+				"Authorization": "Basic " + pce_authorization
+			};
+			var midServer = pceConfig.mid_server.name + "";
+			
+			// Make API call
+			returnDict["response"] = this.makeRestCall("IllumioFetchLabelDimensions", "GetLabelDimensions", stringParams, headers, {}, midServer);
+			returnDict["statusCode"] = returnDict["response"].getStatusCode();
+			returnDict["status"] = true;
+		} catch (exc) {
+			gs.error("Exception while making the API call to get the label dimensions for PCE configuration " + pceConfig.getValue("name") + ". Error - " + exc);
+		}
+		return returnDict;
+	},
+	
+	/**
+	 * Create/Update the label dimension record with the provided details
+	 * @param {dict} contains the label dimension object from the API response obtained from Illumio PCE
+	 * @param {object} GlideRecord of PCE Configuration
+	 */
+	createOrUpdateLabelDimensionRecord: function(details, pceConfig) {
+		try {
+			var labelDimensionGr = new GlideRecord("x_illu2_illumio_illumio_label_dimension");
+			labelDimensionGr.addQuery("key", details["key"]);
+			labelDimensionGr.addQuery("illumio_pce_configuration", pceConfig.getUniqueValue());
+			labelDimensionGr.query();
+			if(!labelDimensionGr.next()) {
+				labelDimensionGr.initialize();
+				labelDimensionGr.setValue("key", details["key"]);
+				labelDimensionGr.setValue("illumio_pce_configuration", pceConfig.getUniqueValue());
+			}
+			labelDimensionGr.setValue("display_name_singular", details["display_name"]);
+			labelDimensionGr.setValue("display_name_plural", details["display_info"]["display_name_plural"]);
+			labelDimensionGr.setValue("href", details["href"]);
+			labelDimensionGr.setValue("label_type_initial", details["display_info"]["initial"]);
+			labelDimensionGr.setValue("isdeleted", false);
+
+			labelDimensionGr.setWorkflow(false);
+			labelDimensionGr.update();
+		} catch (exc) {
+			gs.warn("Error occured while creating/updating the record for Label Dimension " + details["key"] + ". ");
+		}
+	},
+	
+	/**
+     * Fetch the details of label dimensions from PCE
+     * @returns {Boolean} true if the request was successfull else false
```

Comment:
The method's return says boolean but this method doesnt return boolean. Pls check this